jvm/src/main/scala/eval/ParallelInterpreter.scala

```diff
@@ -0,0 +1,58 @@
+package com.azavea.maml.eval
+
+import com.azavea.maml.ast._
+import com.azavea.maml.error._
+import com.azavea.maml.eval.directive._
+
+import cats._
+import cats.implicits._
+import cats.data.Validated._
+import cats.data.{NonEmptyList => NEL, _}
+import cats.effect.{Concurrent, Fiber}
+
+import scala.reflect.ClassTag
+
+class ParallelInterpreter[F[_]](directives: List[Directive])(
+    implicit Conc: Concurrent[F]
+) extends Interpreter[F] {
+  def apply(exp: Expression): F[Interpreted[Result]] = {
+    val children = evalInF(exp)
+    val out = children map {
```

Comment:
It occurs to me that this should construct a series of `F` values to be evaluated with a `parSequence` (or something quite like that) for true parallelism. Here's the typeclass I'd expect to be implicitly guaranteed: https://typelevel.org/cats/typeclasses/parallel.html

I think what you've got here buys concurrency (which might have desirable performance characteristics for us on http4s) but not parallelism.

Reply:
This is a good point -- I will rename this interpreter and also add a parallel interpreter, since there's a good chance that concurrency / parallelism might be more desirable in specific settings.

Reply:
Two new commits:

769e41f changes the existing interpreter to be named `ConcurrentInterpreter`, and 7024414 adds a `ParallelIntepreter` with I think the minimum set of typeclass constraints